<!doctype html><html lang="zh-hans"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>十种改变科学的计算机代码 Ten computer codes that transformed science</title><link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css" integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous"><link rel="stylesheet" href="/img/css.css?random="><link data-rh="true" rel="icon" href="/img/favicon.ico"/><script data-ad-client="ca-pub-6067137220025946" async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script><script type="text/javascript" src="https://platform-api.sharethis.com/js/sharethis.js#property=5effb96910009800120b8d4d&product=inline-share-buttons" async="async"></script>
<script>var _hmt = _hmt || [];(function() {var hm = document.createElement("script");hm.src = "https://hm.baidu.com/hm.js?03c1a0f31299b4a2fbb83c34d6beaac9";var s = document.getElementsByTagName("script")[0]; s.parentNode.insertBefore(hm, s);})();</script></head><body><div id="my_header"><div class="container"><nav class="navbar navbar-expand-lg"><a class="navbar-brand" href="/"><img alt="diglog" src="/img/logo.v1.gif" class="rounded-sm"></a><button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation"><span class="navbar-toggler-icon"></span></button><div class="collapse navbar-collapse" id="navbarNavAltMarkup"><div class="navbar-nav"></div></div></nav></div></div><div class="container"><div id="my_content"><h1 class="page_narrow">Ten computer codes that transformed science<br/>十种改变科学的计算机代码 </h1><div class="row"><div class="col-lg-12 col-12"><div class="my_story_list_item shadow p-3 mb-5 bg-white rounded"><div class="story_page_pub_time page_narrow">2021-01-20 23:47:19</div><div class="story_img_container"><a href="http://img2.diglog.com/img/2021/1/54a850a53c64b3d875a79d976ad4602f.jpg"><img src="http://img2.diglog.com/img/2021/1/54a850a53c64b3d875a79d976ad4602f.jpg" class="img-fluid my_story_img" onerror="this.style.display='none'"></a></div><div class="page_narrow text-break page_content"><p>In 2019, the Event Horizon Telescope team gave the world the first glimpse of what a black hole actually looks like. But the image of a glowing, ring-shaped object that the group unveiled wasn’t a conventional photograph. It was computed — a mathematical transformation of data captured by radio telescopes in the United States, Mexico, Chile, Spain and the South Pole  1. The team released the programming code it used to accomplish that feat alongside the articles that documented its findings, so the scientific community could see — and build on — what it had done.</p><p>在2019年，Event Horizo​​n Telescope团队首次向世界展示了黑洞的实际外观。但是该小组展示的一个发光的环形物体的图像不是传统的照片。它是经过计算的，是对美国，墨西哥，智利，西班牙和南极1台射电望远镜所捕获数据的数学转换。该团队与记录其发现的文章一起发布了用于实现这一壮举的编程代码，因此科学界可以看到并在此基础上继续前进。</p><p> It’s an increasingly common pattern. From astronomy to zoology, behind every great scientific finding of the modern age, there is a computer. Michael Levitt, a computational biologist at Stanford University in California who won a share of the 2013 Nobel Prize in Chemistry for his work on computational strategies for modelling chemical structure, notes that today’s laptops have about 10,000 times the memory and clock speed that his lab-built computer had in 1967, when he began his prizewinning work. “We really do have quite phenomenal amounts of computing at our hands today,” he says. “Trouble is, it still requires thinking.”</p><p> 这是一种越来越普遍的模式。从天文学到生态学，现代每一个伟大的科学发现背后都有一台计算机。加利福尼亚州斯坦福大学的计算生物学家迈克尔·莱维特（Michael Levitt）因其在对化学结构进行建模的计算策略方面的研究而获得了2013年诺贝尔化学奖，他指出，当今的笔记本电脑的存储和时钟速度大约是他实验室的10,000倍1967年，当他开始他的获奖工作时，内置计算机就开始了。他说：“今天，我们确实确实掌握了相当数量的计算。” “问题是，它仍然需要思考。”</p><p> Enter the scientist-coder. A powerful computer is useless without software capable of tackling research questions — and researchers who know how to write it and use it. “Research is now fundamentally connected to software,” says Neil Chue Hong, director of the Software Sustainability Institute, headquartered in Edinburgh, UK, an organization dedicated to improving the development and use of software in science. “It permeates every aspect of the conduct of research.”</p><p> 输入科学家编码。如果没有能够解决研究问题的软件以及知道如何编写和使用它的研究人员，那么功能强大的计算机将毫无用处。 “研究现在已经与软件建立了根本联系，”总部位于英国爱丁堡的软件可持续发展研究所所长Neil Chue Hong说，该组织致力于改善科学软件的开发和使用。 “它渗透到研究行为的各个方面。”</p><p> Scientific discoveries rightly get top billing in the media. But  Nature this week looks behind the scenes, at the key pieces of code that have transformed research over the past few decades.</p><p> 科学发现理应获得媒体的关注。但是本周的《自然》杂志是幕后花絮，在过去几十年中，这些关键代码改变了研究。</p><p> Although no list like this can be definitive, we polled dozens of researchers over the past year to develop a diverse line-up of ten software tools that have had a big impact on the world of science. You can weigh in on our choices at the end of the story.</p><p> 尽管没有像这样的清单可以确定，但是在过去的一年中，我们对数十名研究人员进行了调查，以开发出十种对科学界产生重大影响的软件工具的多样化阵容。在故事的结尾，您可以权衡我们的选择。</p><p>  The first modern computers weren’t user-friendly. Programming was literally done by hand, by connecting banks of circuits with wires. Subsequent machine and assembly languages allowed users to program computers in code, but both still required an intimate knowledge of the computer’s architecture, putting the languages out of reach of many scientists.</p><p>  最初的现代计算机不是用户友好的。编程实际上是通过用电线连接电路组来手工完成的。随后的机器语言和汇编语言允许用户使用代码对计算机进行编程，但是这两种语言都仍需要对计算机的体系结构有深入的了解，这使得许多科学家无法使用这些语言。</p><p> That changed in the 1950s with the development of symbolic languages — in particular the ‘formula translation’ language Fortran, developed by John Backus and his team at IBM in San Jose, California. Using Fortran, users could program computers using human-readable instructions, such as  x = 3 + 5. A compiler then turned such directions into fast, efficient machine code.</p><p> 随着符号语言的发展，这种情况在1950年代发生了变化，特别是John Backus和他在加利福尼亚州圣何塞的IBM团队开发的“公式翻译”语言Fortran。使用Fortran，用户可以使用诸如x instructions = 3 + 5之类的可读指令对计算机进行编程，然后编译器将这些指令转换为快速，高效的机器代码。 </p><p>  It still wasn’t easy: in the early days, programmers used punch cards to input code, and a complex simulation might require tens of thousands of them. Still, says Syukuro Manabe, a climatologist at Princeton University in New Jersey, Fortran made programming accessible to researchers who weren’t computer scientists. “For the first time, we were able to program [the computer] by ourselves,” Manabe says. He and his colleagues used the language to develop one of the first successful climate models.</p><p>这仍然不容易：在早期，程序员使用打孔卡来输入代码，而复杂的仿真可能需要成千上万张。新泽西州普林斯顿大学的气候学家Syukuro Manabe说，Fortran使得非计算机科学家的研究人员可以访问程序。 “第一次，我们能够自己对[计算机]进行编程，” Manabe说。他和他的同事使用该语言开发了第一个成功的气候模型之一。</p><p> Now in its eighth decade, Fortran is still widely used in climate modelling, fluid dynamics, computational chemistry — any discipline that involves complex linear algebra and requires powerful computers to crunch numbers quickly. The resulting code is fast, and there are still plenty of programmers who know how to write it. Vintage Fortran code bases are still alive and kicking in labs and on supercomputers worldwide. “Old-time programmers knew what they were doing,” says Frank Giraldo, an applied mathematician and climate modeller at the Naval Postgraduate School in Monterey, California. “They were very mindful of memory, because they had so little of it.”</p><p> 如今，在其第八个十年中，Fortran仍被广泛用于气候建模，流体动力学，计算化学等涉及复杂线性代数且需要强大计算机快速处理数字的学科。生成的代码很快，并且仍然有很多程序员知道如何编写它。在全球的实验室和超级计算机上，Vintage Fortran代码库仍然有效。 “资深程序员知道他们在做什么，”加利福尼亚蒙特雷海军研究生院的应用数学家和气候建模师Frank Giraldo说。 “他们非常注意记忆，因为他们的记忆很少。”</p><p>  When radioastronomers scan the sky, they capture a cacophony of complex signals changing with time. To understand the nature of those radio waves, they need to see what those signals look like as a function of frequency. A mathematical process called a Fourier transform allows researchers to do that. The problem is that it’s inefficient, requiring  N 2 calculations for a data set of size  N.</p><p>  当射电天文学家扫描天空时，他们捕获了随时间变化的复杂信号的杂音。为了了解这些无线电波的性质，他们需要查看这些信号作为频率函数的样子。研究人员可以通过称为傅立叶变换的数学过程来做到这一点。问题是效率低下，需要对大小为N的数据集进行N 2次计算。</p><p> In 1965, US mathematicians James Cooley and John Tukey worked out a way to accelerate the process. Using recursion, a ‘divide and conquer’ programming approach in which an algorithm repeatedly reapplies itself, the fast Fourier transform (FFT) simplifies the problem of computing a Fourier transform to just  N log 2( N) steps. The speed improves as  N grows. For 1,000 points, the speed boost is about 100-fold; for 1 million points, it’s 50,000-fold.</p><p> 1965年，美国数学家James Cooley和John Tukey提出了一种加快这一过程的方法。使用递归（一种“分而治之”的编程方法，其中算法会重复应用自身），快速傅立叶变换（FFT）简化了将傅立叶变换计算为N log 2（N）个步骤的问题。速度随着N的增长而提高。对于1,000点，速度提升约为100倍；一百万点，是五万倍</p><p> The ‘discovery’ was actually a rediscovery — the German mathematician Carl Friedrich Gauss worked it out in 1805, but he never published it, says Nick Trefethen, a mathematician at the University of Oxford, UK. But Cooley and Tukey did, opening applications in digital signal processing, image analysis, structural biology and more. “It’s really one of the great events in applied mathematics and engineering,” Trefethen says. FFT has been implemented many times in code. One popular option is called FFTW, the ‘fastest Fourier transform in the west’.</p><p> 英国牛津大学数学家尼克·特雷菲森（Nick Trefethen）说，“发现”实际上是一次重新发现-德国数学家卡尔·弗里德里希·高斯（Carl Friedrich Gauss）于1805年完成了该发现，但他从未发表过。但是Cooley和Tukey做到了，打开了在数字信号处理，图像分析，结构生物学等领域的应用程序。 Trefethen说：“这确实是应用数学和工程学领域的重大事件之一。” FFT已经在代码中实现了很多次。一种流行的选择称为FFTW，即“西方最快的傅立叶变换”。</p><p>  Paul Adams, who directs the molecular biophysics and integrated bioimaging division at Lawrence Berkeley National Laboratory in California, recalls that when he refined the structure of the bacterial protein GroEL in 1995  2, the calculation took “many, many hours, if not days”, even with the FFT and a supercomputer. “Trying to do those without the FFT, I don’t even know how we would have done that realistically,” he says. “It would have just taken forever.”</p><p>  保罗·亚当斯（Paul Adams）负责加利福尼亚州劳伦斯·伯克利国家实验室的分子生物物理学和整合生物成像部门，他回忆说，当他在1995年2月改进细菌蛋白GroEL的结构时，计算花费了“很多很多小时，甚至不是几天”，即使使用FFT和超级计算机。他说：“尝试在不使用FFT的情况下，我什至不知道我们将如何现实地做到这一点。” “那将是永远的。”</p><p>  Databases are such a seamless component of scientific research today that it can be easy to overlook the fact that they are driven by software. In the past few decades, these resources have ballooned in size and shaped many fields, but perhaps nowhere has that transformation been more dramatic than in biology.</p><p>  数据库是当今科学研究的一个无缝组成部分，因此可以很容易地忽略它们由软件驱动的事实。在过去的几十年中，这些资源的规模激增，并影响了许多领域，但也许没有哪一种变化比生物学更具戏剧性。 </p><p> Today’s massive genome and protein databases have their roots in the work of Margaret Dayhoff, a bioinformatics pioneer at the National Biomedical Research Foundation in Silver Spring, Maryland. In the early 1960s, as biologists worked to tease apart proteins’ amino acid sequences, Dayhoff began collating that information in search of clues into evolutionary relationships between different species. Her  Atlas of Protein Sequence and Structure, first published in 1965 with three co-authors, described what was then known of the sequences, structures and similarities of 65 proteins. The collection was the first that “was not tied to a specific research question”, historian Bruno Strasser wrote in 2010  3. And it encoded its data in punch cards, which made it possible to expand the database and search it.</p><p>当今庞大的基因组和蛋白质数据库源于玛格丽特·戴霍夫（Margaret Dayhoff）的工作，他是马里兰州银泉市国家生物医学研究基金会的生物信息学先驱。在1960年代初期，随着生物学家努力弄清蛋白质的氨基酸序列，Dayhoff开始整理这些信息，以寻找不同物种之间进化关系的线索。她的《蛋白质序列和结构图谱》于1965年由三位合著者首次发表，描述了当时对65种蛋白质的序列，结构和相似性的了解。历史学家布鲁诺·斯特拉瑟（Bruno Strasser）在2010年3收集了第一个“不依赖于特定研究问题”的集合。它用打孔卡对数据进行编码，从而有可能扩展数据库并进行搜索。</p><p> Other computerized biological databases followed. The Protein Data Bank, which today details more than 170,000 macromolecular structures, went live in 1971. Russell Doolittle, an evolutionary biologist at the University of California, San Diego, created another protein database called Newat in 1981. And 1982 saw the release of the database that would become GenBank, the DNA archive maintained by the US National Institutes of Health.</p><p> 随后是其他计算机生物数据库。 1971年启用了蛋白质数据库，该数据库今天详细描述了170,000个以上的大分子结构。加利福尼亚大学圣地亚哥分校的进化生物学家罗素·杜利特尔（Russell Doolittle）于1981年创建了另一个名为Newat的蛋白质数据库。1982年，该数据库得以发布。该数据库将成为GenBank，即由美国国立卫生研究院维护的DNA档案。</p><p>  Such resources proved their worth in July 1983, when separate teams led by Michael Waterfield, a protein biochemist at the Imperial Cancer Research Fund in London, and Doolittle independently reported a similarity between the sequences of a particular human growth factor and a protein in a virus that causes cancer in monkeys. The observation suggested a mechanism for oncogenesis-by-virus — that by mimicking a growth factor, the virus induces uncontrolled growth of cells  4. “That set the light bulb off in some of the minds of biologists who were not into computers and statistics,” says James Ostell, former director of the US National Center for Biotechnology Information (NCBI): “We can understand something about cancer from comparing sequences.”</p><p>  1983年7月，这种资源证明了自己的价值，当时由伦敦帝国癌症研究基金会的蛋白质生物化学家迈克尔·沃特菲尔德（Michael Waterfield）领导的独立小组和杜利特尔（Doolittle）独立报告了特定人类生长因子的序列与病毒中的蛋白质之间的相似性。导致猴子患癌症。该观察结果提出了一种由病毒致癌的机制-通过模仿生长因子，该病毒诱导了细胞4的不受控制的生长。“这使一些生物学家不去研究计算机和统计学的人想到了灯泡，美国国家生物技术信息中心（NCBI）前任主任詹姆斯·奥斯特（James Ostell）说：“我们可以通过比较序列来了解一些有关癌症的知识。”</p><p> Beyond that, Ostell says, the discovery marked “an advent of objective biology”. In addition to designing experiments to test specific hypotheses, researchers could mine public data sets for connections that might never have occurred to those who actually collected the data. That power grows drastically when different data sets are linked together — something NCBI programmers achieved in 1991 with Entrez, a tool that allows researchers to freely navigate from DNA to protein to literature and back.</p><p> Ostell说，除此之外，这项发现标志着“客观生物学的问世”。除了设计用于测试特定假设的实验外，研究人员还可以挖掘公共数据集，以获取与实际收集数据的人可能从未发生过的联系。当将不同的数据集链接在一起时，这种能力急剧增加。1991年，NCBI程序员使用Entrez实现了这一目标，该工具使研究人员可以自由地从DNA导航到蛋白质再到文献再返回。</p><p> Stephen Sherry, current acting director of the NCBI in Bethesda, Maryland, used Entrez as a graduate student. “I remember at the time thinking it was magic,” he says.</p><p> 现任马里兰州贝塞斯达市NCBI的现任导演Stephen Sherry使用Entrez作为研究生。他说：“我当时记得当时以为是魔术。”</p><p>  At the close of the Second World War, computer pioneer John von Neumann began turning computers that a few years earlier had been calculating ballistics trajectories and weapon designs towards the problem of weather prediction. Up until that point, explains Manabe, “weather forecasting was just empirical”, using experience and hunches to predict what would happen next. Von Neumann’s team, by contrast, “attempted to do numerical weather prediction based upon laws of physics”.</p><p>  第二次世界大战结束时，计算机先驱约翰·冯·诺伊曼（John von Neumann）开始将计算机（几年前一直在计算弹道轨迹和武器设计）转向天气预报问题。 Manabe解释说，直到那时，“天气预报只是经验性的”，利用经验和直觉来预测接下来会发生什么。相比之下，冯·诺依曼（Von Neumann）的团队“试图根据物理定律进行数值天气预报”。</p><p> The equations had been known for decades, says Venkatramani Balaji, head of the Modeling Systems Division at the National Oceanographic and Atmospheric Administration’s Geophysical Fluid Dynamics Laboratory in Princeton, New Jersey. But early meteorologists couldn’t solve them practically. To do so required inputting current conditions, calculating how they would change over a short time period, and repeating — a process so time-consuming that the mathematics couldn’t be completed before the weather itself caught up. In 1922, the mathematician Lewis Fry Richardson spent months crunching a six-hour forecast for Munich, Germany. The result, according to one history, was “wildly inaccurate”, including predictions that “could never occur under any known terrestrial conditions”. Computers made the problem tractable.</p><p> 位于新泽西州普林斯顿的美国国家海洋与大气管理局地球物理流体动力学实验室建模系统部门负责人Venkatramani Balaji说，这些方程式已经存在了数十年。但是早期的气象学家几乎无法解决这些问题。为此，需要输入当前条件，计算它们在短时间内的变化情况，然后重复进行-这一过程非常耗时，以至于无法在天气赶上时完成数学运算。 1922年，数学家刘易斯·弗莱·理查森（Lewis Fry Richardson）花了几个月的时间对德国慕尼黑进行为时6小时的预报。根据一个历史，结果是“非常不准确的”，包括“在任何已知的陆地条件下都不可能发生”的预测。计算机使问题变得易于处理。 </p><p> In the late 1940s, von Neumann established his weather-prediction team at the Institute for Advanced Study at Princeton. In 1955, a second team — the Geophysical Fluid Dynamics Laboratory — began work on what he called “the infinite forecast” — that is, climate modelling.</p><p>在1940年代后期，冯·诺伊曼（von Neumann）在普林斯顿高级研究所成立了天气预报小组。 1955年，第二个小组-地球物理流体动力学实验室-开始研究他所谓的“无限预报”（即气候建模）。</p><p> Manabe, who joined the climate modelling team in 1958, set to work on atmospheric models; his colleague Kirk Bryan addressed those for the ocean. In 1969, they successfully combined the two, creating what  Nature in 2006 called  a “milestone” in scientific computing.</p><p> Manabe于1958年加入气候建模小组，致力于大气模型的研究。他的同事柯克·布莱恩（Kirk Bryan）向海洋致辞。 1969年，他们成功地将两者结合在一起，创造了自然界在2006年称之为科学计算的“里程碑”。</p><p> Today’s models can divide the planet’s surface into squares measuring 25 × 25 kilometres, and the atmosphere into dozens of levels. By contrast, Manabe and Bryan’s combined ocean–atmosphere model  5 used 500-km squares and 9 levels, and covered just one-sixth of the globe. Still, says Balaji, “that model did a great job”, allowing the team to test for the first time the impact of rising carbon dioxide levels  in silico.</p><p> 如今的模型可以将地球的表面划分为25××25公里的正方形，并将大气划分为数十个层次。相比之下，Manabe和Bryan的综合海洋-大气模型5使用了500公里的正方形和9个水平，仅覆盖了全球的六分之一。 Balaji说，“这种模式做得很好”，这使团队能够首次测试硅粉中二氧化碳水平升高的影响。</p><p>  Scientific computing typically involves relatively simple mathematical operations using vectors and matrices. There are just a lot of them. But in the 1970s, there was no universally agreed set of computational tools for performing such operations. As a result, programmers working in science would spend their time devising efficient code to do basic mathematics rather than focusing on scientific questions.</p><p>  科学计算通常涉及使用向量和矩阵的相对简单的数学运算。只有很多。但是在1970年代，还没有一套普遍认可的用于执行此类操作的计算工具。结果，从事科学工作的程序员将花时间设计高效的代码来进行基本数学运算，而不是专注于科学问题。</p><p> What the programming world needed was a standard. In 1979, it got one: Basic Linear Algebra Subprograms, or BLAS  6. The standard, which continued to evolve up to 1990, defined dozens of fundamental routines for vector and, later, matrix mathematics.</p><p> 编程世界需要的是一个标准。在1979年，它得到了一个：基本线性代数子程序，即BLAS6。该标准一直持续发展到1990年，为矢量和后来的矩阵数学定义了数十种基本例程。</p><p> In effect, BLAS reduced matrix and vector mathematics to a basic unit of computation as fundamental as addition and subtraction, says Jack Dongarra, a computer scientist at the University of Tennessee in Knoxville who was a member of the BLAS development team.</p><p> 田纳西大学诺克斯维尔分校计算机科学家杰克·东纳拉（Jack Dongarra）说，实际上，BLAS将矩阵和矢量数学简化为基本的计算单位，而加减法则是其中的基础。</p><p>  BLAS was “probably the most consequential interface to be defined for scientific computing”, says Robert van de Geijn, a computer scientist at the University of Texas at Austin. In addition to providing standardized names for common functions, researchers could be sure BLAS-based code would work in the same manner on any computer. The standard also enabled computer manufacturers to optimize BLAS implementations for speedy operation on their hardware.</p><p>  德克萨斯大学奥斯汀分校的计算机科学家Robert van de Geijn说，BLAS“可能是为科学计算定义的最重要的接口”。除了为通用功能提供标准化名称外，研究人员还可以确保基于BLAS的代码可以在任何计算机上以相同的方式工作。该标准还使计算机制造商能够优化BLAS实现，以在其硬件上快速运行。 </p><p> More than 40 years on, BLAS represents the heart of the scientific computing stack, the code that makes scientific software tick. Lorena Barba, a mechanical and aerospace engineer at George Washington University in Washington DC, calls it “the machinery inside five layers of code”.</p><p>40多年来，BLAS代表了科学计算堆栈的心脏，这是使科学软件产生变化的代码。华盛顿特区乔治华盛顿大学的机械和航空工程师Lorena Barba称其为“五层代码内的机械”。</p><p>   In the early 1980s, programmer Wayne Rasband was working with a brain-imaging lab at the US National Institutes of Health in Bethesda, Maryland. The team had a scanner to digitize X-ray films, but no way to display or analyse them on their computer. So Rasband wrote a program to do just that.</p><p>   在1980年代初期，程序员Wayne Rasband在马里兰州贝塞斯达的美国国立卫生研究院的脑成像实验室工作。该团队使用扫描仪将X射线胶片数字化，但无法在计算机上显示或分析它们。因此，拉斯班德（Rasband）编写了一个程序来做到这一点。</p><p> The program was specifically designed for a US$150,000 PDP-11 minicomputer — a rack-mounted, decidedly non-personal computer. Then, in 1987, Apple released its Macintosh II, a friendlier and much more affordable option. “It seemed obvious to me that that would work a lot better as a kind of laboratory image analysis system,” Rasband says. He ported his software to the new platform and rebranded it, seeding an image-analysis ecosystem.</p><p> 该程序是专门为价值15万美元的PDP-11微型计算机设计的，PDP-11微型计算机是机架安装的非个人计算机。然后，在1987年，苹果发布了它的Macintosh II，这是一个更友好，更实惠的选择。 “对我来说，显而易见的是，作为一种实验室图像分析系统，它会更好得多，” Rasband说。他将软件移植到新平台上并重新命名，从而建立了图像分析生态系统。</p><p> NIH Image and its descendants empowered researchers to view and quantify just about any image, on any computer. The software family includes ImageJ, a Java-based version that Rasband wrote for Windows and Linux users, and Fiji, a distribution of ImageJ developed by Pavel Tomancak’s group at the Max Planck Institute of Molecular Cell Biology and Genetics in Dresden, Germany, that includes key plug-ins. “ImageJ is certainly the most foundational tool that we have,” says Beth Cimini, a computational biologist who works on the Imaging Platform of the Broad Institute in Cambridge, Massachusetts. “I’ve literally never spoken to a biologist who has used a microscope but not ImageJ or its offshoot project, Fiji.”</p><p> NIH Image及其后代使研究人员可以在任何计算机上查看和量化几乎任何图像。该软件系列包括Rasband为Windows和Linux用户编写的基于Java的版本ImageJ，以及由德国德累斯顿马克斯·普朗克分子细胞生物学与遗传学研究所的Pavel Tomancak小组开发的ImageJ发行版Fiji，其中包括关键插件。 “ ImageJ无疑是我们拥有的最基础的工具，”在马萨诸塞州剑桥市Broad Institute的成像平台上工作的计算生物学家Beth Cimini说。 “从字面上看，我从未与使用过显微镜但没有使用ImageJ或其分支项目斐济的生物学家说话。”</p><p>  That’s partly because these tools are free, Rasband says. But it’s also because it’s easy for users to customize the tool to their needs, says Kevin Eliceiri, a biomedical engineer at the University of Wisconsin–Madison, whose team has taken the lead on ImageJ development since Rasband’s retirement. ImageJ features a deceptively simple, minimalist user interface that has remained largely unchanged since the 1990s. Yet the tool is infinitely extensible thanks to its built-in macro recorder (which allows a user to save workflows by recording sequences of mouse clicks and menu selections), extensive file-format compatibility and flexible plug-in architecture. “Hundreds of people” have contributed plug-ins, says Curtis Rueden, the programming lead in Eliceiri’s group. These additions have greatly expanded the toolset for researchers, with functions to track objects over time in videos or automatically identify cells, for instance.</p><p>  Rasband说，部分原因是这些工具是免费的。威斯康星大学麦迪逊分校的生物医学工程师凯文·埃里斯里（Kevin Eliceiri）说，但这也是因为用户可以轻松地根据自己的需求定制该工具，该团队自Rasband退休以来一直负责ImageJ的开发。 ImageJ具有看似简单，极简的用户界面，自1990年代以来，该界面基本上保持不变。然而，由于其内置的宏记录器（允许用户通过记录鼠标单击和菜单选择的顺序来保存工作流），广泛的文件格式兼容性和灵活的插件体系结构，该工具可以无限扩展。 Eliceiri小组的编程负责人Curtis Rueden说，“成百上千的人”贡献了插件。这些新增功能极大地扩展了研究人员的工具集，例如具有随时间推移跟踪视频中对象的功能或自动识别单元格的功能。</p><p> “The point of the program isn’t to be the be-all and end-all,” Eliceiri says, “it’s to serve the purpose of its users. And unlike Photoshop and other programs, ImageJ can be whatever you want it to be.”</p><p> Eliceiri说：“该计划的重点不是成为最终目标，而是为了达到用户的目的。与Photoshop和其他程序不同，ImageJ可以随心所欲。</p><p>  There might be no better indicator of cultural relevance than for a software name to become a verb. For search, think Google. And for genetics, think BLAST.</p><p>  可能没有比将软件名称变成动词更好的文化相关性指标。对于搜索，请考虑使用Google。对于遗传学，请考虑一下BLAST。 </p><p> Evolutionary changes are etched into molecular sequences as substitutions, deletions, gaps and rearrangements. By searching for similarities between sequences — particularly among proteins — researchers can discover evolutionary relationships and gain insight into gene function. The trick is to do so quickly and comprehensively across rapidly ballooning databases of molecular information.</p><p>进化变化被刻蚀为分子序列，如取代，缺失，缺口和重排。通过搜索序列之间的相似性，尤其是蛋白质之间的相似性，研究人员可以发现进化关系并深入了解基因功能。诀窍是在迅速增加的分子信息数据库中快速而全面地进行操作。</p><p> Dayhoff provided one crucial piece of the puzzle in 1978. She devised a ‘point accepted mutation’ matrix that allowed researchers to score the relatedness of two proteins based not only on how similar their sequences are, but also on the evolutionary distance between them.</p><p> Dayhoff在1978年提出了一个至关重要的难题。她设计了一个“点接受突变”矩阵，使研究人员不仅可以根据序列的相似程度，还可以根据它们之间的进化距离来对两种蛋白质的相关性进行评分。</p><p> In 1985, William Pearson at the University of Virginia in Charlottesville and David Lipman at the NCBI introduced FASTP, an algorithm that combined Dayhoff’s matrix with the ability to perform rapid searches.</p><p> 1985年，夏洛茨维尔的弗吉尼亚大学的William Pearson和NCBI的David Lipman引入了FASTP，该算法结合了Dayhoff的矩阵和执行快速搜索的能力。</p><p> Years later, Lipman, along with Warren Gish and Stephen Altschul at the NCBI, Webb Miller at Pennsylvania State University in University Park, and Gene Myers at the University of Arizona, Tucson, developed an even more powerful refinement: the Basic Local Alignment Search Tool (BLAST). Released in 1990, BLAST combined the search speed required to handle fast-growing databases with the ability to pick up matches that were more evolutionarily distant. At the same time, the tool could calculate how likely it is that those matches occurred by chance.</p><p> 多年后，Lipman与NCBI的Warren Gish和Stephen Altschul，大学公园的宾夕法尼亚州立大学的Webb Miller以及图森的亚利桑那大学的Gene Myers一起开发了更强大的改进方法：基本局部比对搜索工具（爆破）。 BLAST于1990年发布，将处理快速增长的数据库所需的搜索速度与能够找到进化距离更远的匹配项的能力相结合。同时，该工具可以计算出这些匹配是偶然发生的可能性。</p><p> The result was incredibly fast, Altschul says. “You could put in your search, take one sip of coffee, and your search would be done.” But more importantly, it was easy to use. In an era when databases were updated by post, Gish established an e-mail system and later a web-based architecture that allowed users to run searches on the NCBI computers remotely, thus ensuring their results were always up-to-date.</p><p> 结果非常快，Altschul说。 “您可以进行搜索，喝一口咖啡，然后搜索便会完成。”但更重要的是，它易于使用。在通过邮寄更新数据库的时代，Gish建立了电子邮件系统，后来建立了基于Web的体系结构，该体系结构允许用户在NCBI计算机上远程运行搜索，从而确保其结果始终是最新的。</p><p> The system gave the then-budding field of genome biology a transformative tool, says Sean Eddy, a computational biologist at Harvard University in Cambridge, Massachusetts — a way to work out what unknown genes might do on the basis of the genes they were related to. And for sequencing labs everywhere, it provided a clever neologism: “It’s just one of these things that became a verb,” Eddy says. “You just talked about BLASTing your sequences.”</p><p> 马萨诸塞州剑桥市哈佛大学的计算生物学家肖恩·埃迪说，该系统为当时处于萌芽状态的基因组生物学领域提供了一种转化工具，这是一种根据未知基因可能与之相关的基因找出未知基因可能做什么的方法。 。对于世界各地的测序实验室，它提供了一种巧妙的新词：“这只是变成动词的其中之一，”埃迪说。 “您刚刚谈论了爆破序列。”</p><p>  In the late 1980s, high-energy physicists routinely sent physical copies of their submitted manuscripts to colleagues by post for comment and as a courtesy — but only to a select few. “Those lower in the food chain relied on the beneficence of those on the A-list, and aspiring researchers at non-elite institutions were frequently out of the privileged loop entirely,” wrote physicist Paul Ginsparg in 2011  7.</p><p>  在1980年代后期，高能物理学家通常将其提交的手稿的物理副本以邮寄方式寄给同事，以征求评论和礼貌，但只发给了少数几个人。物理学家保罗·金斯帕格（Paul Ginsparg）在2011年7月写道：“食物链中较低的那些人依靠的是A名单上那些人的仁慈，非精英机构的有抱负的研究人员经常完全不在特权之列。” </p><p> In 1991, Ginsparg, then at Los Alamos National Laboratory in New Mexico, wrote an e-mail autoresponder to level the playing field. Subscribers received daily lists of preprints, each associated with an article identifier. With a single e-mail, users across the world could submit or retrieve an article from the lab’s computer system, get lists of new articles or search by author or title.</p><p>1991年，当时在新墨西哥州洛斯阿拉莫斯国家实验室的金斯帕格（Ginsparg）撰写了一封电子邮件自动回复程序，以公平地竞争环境。订户每天收到预印本的清单，每个清单都与文章标识符相关联。只需一封电子邮件，全世界的用户都可以从实验室的计算机系统提交或检索文章，获取新文章的列表，或者按作者或标题进行搜索。</p><p> Ginsparg’s plan was to retain articles for three months, and to limit content to the high-energy physics community. But a colleague convinced him to retain the articles indefinitely. “That was the moment it transitioned from bulletin board to archive,” he says. And papers flooded in from much farther afield than Ginsparg’s own discipline. In 1993, Ginsparg migrated the system to the Internet, and in 1998 he gave it the name it goes by today: arXiv.org.</p><p> 金斯帕格（Ginsparg）的计划是将文章保留三个月，并将内容限制在高能物理领域。但是一位同事说服他无限期保留这些文章。他说：“那是从公告栏过渡到存档的那一刻。”而且，论文的出版范围比金斯帕格所在学科要远得多。 1993年，金斯帕格（Ginsparg）将系统迁移到了互联网，并在1998年将其命名为arXiv.org。</p><p> Now in its thirtieth year, arXiv houses some 1.8 million preprints — all available for free — and attracts more than 15,000 submissions and some 30 million downloads per month. “It’s not hard to see why the arXiv is such a popular service,” the editors of  Nature Photonics wrote  8 a decade ago on the occasion of the site’s twentieth anniversary: “The system provides researchers with a fast and convenient way to plant a flag that shows what they did and when, avoiding the hassle and time required for peer review at a conventional journal.”</p><p> 如今已是第30年，arXiv拥有约180万张预印本（全部免费提供），每月吸引超过15,000份投稿和约3000万次下载。 Nature Photonics的编辑在十年前网站成立20周年之际写道：“不难看出arXiv为何如此受欢迎的服务，”该系统为研究人员提供了一种快速便捷的方法来制作国旗可以显示他们的工作和时间，从而避免了传统期刊上同行评审所带来的麻烦和时间。”</p><p>  The site’s success catalysed a boom in sister archives in biology, medicine, sociology and other disciplines. The impact can be seen today in tens of thousands of preprints that have been published on the virus SARS-CoV-2.</p><p>  该网站的成功催生了生物学，医学，社会学和其他学科的姊妹档案激增。今天，已经在SARS-CoV-2病毒上发布了成千上万的预印本，可以看到这种影响。</p><p> “It’s gratifying to see a methodology, considered heterodox outside of the par</p><p> “很高兴看到一种方法，被认为是同等水平之外的异质</p><p>......</p><p>...... </p></div><div id="story_share_this"><div class="sharethis-inline-share-buttons"></div></div><div class="text-break sotry_link page_narrow"><a target="_blank" href="https://www.nature.com/articles/d41586-021-00075-2">https://www.nature.com/articles/d41586-021-00075-2</a></div><div class="story_tags page_narrow"><button type="button" class="btn btn-light my_tag"><a href="/tag/代码/">#代码</a></button><button type="button" class="btn btn-light my_tag"><a href="/tag/科学/">#科学</a></button><button type="button" class="btn btn-light my_tag"><a href="/tag/computer/">#computer</a></button></div></div><div class="my_movie_list_item shadow p-3 mb-5 bg-white rounded"><button type="button" class="btn btn-link my_tag"><a href="/tag/web2.0/">#web2.0</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/google/">#google</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/设计/">#设计</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/创意/">#创意</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/摄影/">#摄影</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/图片/">#图片</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/游戏/">#游戏</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/软件/">#软件</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/手机/">#手机</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/视频/">#视频</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/广告/">#广告</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/iphone/">#iphone</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/网站/">#网站</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/免费/">#免费</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/下载/">#下载</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/windows/">#windows</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/微软/">#微软</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/苹果/">#苹果</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/apple/">#apple</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/firefox/">#firefox</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/blog/">#blog</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/音乐/">#音乐</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/博客/">#博客</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/wordpress/">#wordpress</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/恶搞/">#恶搞</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/qq/">#qq</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/艺术/">#艺术</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/web/">#web</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/工具/">#工具</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/分享/">#分享</a></button></div></div></div><div id="my_footer"><div class=""><a href="/tags/">tags</a> <a href="/users/">users</a></div>&copy; 2020 diglog.com </div></div></body></html>