<!doctype html><html lang="zh-hans"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>SARS-COV-2的新出现可塑性 The emerging plasticity of SARS-CoV-2</title><link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css" integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous"><link rel="stylesheet" href="/img/css.css?random="><link data-rh="true" rel="icon" href="/img/favicon.ico"/><script data-ad-client="ca-pub-6067137220025946" async src="https://pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script><script type="text/javascript" src="https://platform-api.sharethis.com/js/sharethis.js#property=5effb96910009800120b8d4d&product=inline-share-buttons" async="async"></script>
<script>var _hmt = _hmt || [];(function() {var hm = document.createElement("script");hm.src = "https://hm.baidu.com/hm.js?03c1a0f31299b4a2fbb83c34d6beaac9";var s = document.getElementsByTagName("script")[0]; s.parentNode.insertBefore(hm, s);})();</script></head><body><div id="my_header"><div class="container"><nav class="navbar navbar-expand-lg"><a class="navbar-brand" href="/"><img alt="diglog" src="/img/logo.v1.gif" class="rounded-sm"></a><button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarNavAltMarkup" aria-controls="navbarNavAltMarkup" aria-expanded="false" aria-label="Toggle navigation"><span class="navbar-toggler-icon"></span></button><div class="collapse navbar-collapse" id="navbarNavAltMarkup"><div class="navbar-nav"></div></div></nav></div></div><div class="container"><div id="my_content"><h1 class="page_narrow">The emerging plasticity of SARS-CoV-2<br/>SARS-COV-2的新出现可塑性 </h1><div class="row"><div class="col-lg-12 col-12"><div class="my_story_list_item shadow p-3 mb-5 bg-white rounded"><div class="story_page_pub_time page_narrow">2021-03-28 00:56:07</div><div class="story_img_container"><a href="http://img2.diglog.com/img/2021/3/ba7a8dd767d27cd3ee4c69db15dad91a.jpg"><img src="http://img2.diglog.com/img/2021/3/ba7a8dd767d27cd3ee4c69db15dad91a.jpg" class="img-fluid my_story_img" onerror="this.style.display='none'"></a></div><div class="page_narrow text-break page_content"><p>Viruses evolve as a result of mutation (misincorporations, insertions or deletions, and recombination) and natural selection for favorable traits such as more efficient viral replication, transmission, and evasion of host defenses. Newly selected traits may be linked in unpredictable ways and raise concern that virus spread and evolution could result in greater virulence (disease severity). The limited diversity of severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2) reported during 2020, ascribed to the 3′-5′ exonuclease proofreading function of nonstructural protein 14 (nsp14), led to the view that vaccines based on a single sequence of the viral spike (S) protein, which mediates host cell entry, would likely generate immune protection to all circulating variants ([ 1 ][1]). However, variants of SARS-CoV-2 with mutations in S have emerged around the world, posing potential challenges for vaccination and antibody-based therapies. The continued spread of SARS-CoV-2 creates the opportunity for accumulation of additional consequential mutations in S and throughout the viral genome.Although SARS-CoV-2 shares high sequence homology with SARS-CoV, which caused the 2002–2004 SARS outbreak, the coronavirus family is diverse in both sequence and in host receptor preference. For example, SARS-CoV-2 and a “common cold” human coronavirus, HCoV-NL63, both recognize angiotensin-converting enzyme 2 (ACE2) as the host cell receptor, but SARS-CoV-2 and HCoV-NL63 belong to different coronavirus genera and have major sequence and structural differences in the receptor-binding domain (RBD) of S, sharing 100 million people, there was limited evidence for evolution of SARS-CoV-2 S. The only notable evolutionary event was the D614G (Asp614→Gly) substitution in S1, which increases ACE2 affinity, leading to higher infectivity and transmissibility. Viral sequences deposited in public databases were mostly obtained from the upper respiratory tract during acute infection, before major immune responses have occurred. Such sequences might not capture the effect of within-host immune selection on viral diversification.Extensive intrahost evolution of SARS-CoV-2 has been reported in at least five individuals with protracted infection because of immune impairment from therapy for hematologic malignancies or autoimmunity ([ 3 ][3]–[ 7 ][4]). They had active SARS-CoV-2 infection for an average of 115 days before clearing the infection or succumbing to COVID-19. Each patient also had at least one convalescent plasma (CP) treatment (intravenous transfusion of blood plasma from a donor who has recovered from COVID-19) and/or monoclonal antibody therapy. Some of these individuals were shedding high titers of SARS-CoV-2 at the time of discharge from hospital or before death, indicating the potential for transmission. SARS-CoV-2 variants from two of these patients had up to fivefold reduction in neutralization sensitivity to CP ([ 3 ][3], [ 7 ][4]). Although these are case studies in immunocompromised individuals, they raise concern because the deletions of amino acids 69 to 70 (Δ69–70), Δ141–144, or Δ242–248 in S1 were observed in four out of the five infections ([ 3 ][3], [ 5 ][5]–[ 7 ][4]); the N501T (Asn501→Thr) or N501Y (Asn501→Tyr) mutations were seen in two out of the five ([ 5 ][5], [ 6 ][6]); and the E484K (Glu484→Lys) and Q493K (Gln493→Lys) mutations in the RBD of one infection also arose in antibody-resistant viruses after in vitro selection.These reports preceded the detection of three major circulating variants—B.1.1.7, B.1.351, and P.1—which all contain at least eight single, nonsynonymous nucleotide changes, including E484K, N501Y, and/or K417N (Lys417→Asn) in the ACE2 interface of the RBD (shown in the illustration). There are also various deletions in the amino (N)-terminal domain (NTD) of S1 in B.1.1.7 and B.1.351 (see the figure). Although most of the mutations in these variants were observed in a minor fraction of SARS-CoV-2 sequences during the first year of the pandemic, including K417N, E484K, and N501Y, there is no evidence to suggest that these variants were created through sequential addition of each substitution during interhost transmission.Because only a few SARS-CoV-2 mutations were in circulation during most of 2020, it is likely that the three major variants are the result of selective pressures and adaptation of the virus during prolonged individual infections and subsequent transmission. All the case reports of individuals with extensive intrahost SARS-CoV-2 evolution indicated that they had been treated with suboptimal neutralizing antibodies (that is, the CP treatment did not neutralize the entire virus population). Whether or not antibody therapy played a role, it is likely that the same variants or variants containing new mutations will continue to emerge in different geographic locations as the result of intrahost selection and subsequent transmission. Indeed, other variants have been reported with multiple mutations in S1, including the lineages B.1.526 (detected in New York) and B.1.429 (which originated in California) containing a substitution in the RBD that is distinct from other variants; and B.1.525 and A.23.1 that are thought to have originated in Nigeria and Uganda, respectively ([ 8 ][7]) (see the figure).The individual phenotypic effects of the mutations in S1 are incompletely understood, but some initial clues are emerging. Substitution at position Asn501 with Thr or Phe increases affinity for ACE2 binding ([ 9 ][8]), and Tyr501 increases infectivity and virulence in a mouse model ([ 10 ][9]). Some circulating variants may have reduced sensitivity to neutralizing antibodies that bind to the RBD directly (attributed to triple substitutions of key amino acids in the RBD at the ACE2-binding interface: Lys417, Glu484, and Asn501) or to the NTD (conformational changes in the NTD are required for ACE2 attachment). More studies to correlate viral genotype and phenotype are needed.It is possible that mutations that reduce neutralizing antibody binding, such as E484K, may require compensatory mutations that restore infectivity, such as N501Y. There appears to be convergent association of mutations such as the triple RBD mutation (Lys417, Glu484, and Asn501) that evolved in two distinct lineages (B.1.351 and P.1). Moreover, E484K was also recently detected with N501Y in the B.1.1.7 lineage ([ 11 ][10]). Δ69–70 in S1 doubled the infectivity of SARS-CoV-2 pseudovirus, implying that the deletion may have been required to compensate for a mutation, D796H (Asp796→His), that reduced antibody neutralization sensitivity at a cost to viral fitness ([ 7 ][4]). The role of compensatory mutations is also supported by the emerging B.1.525 lineage that has both E484K (reduction in antibody sensitivity) and Δ69–70 (compensatory increase in infectivity).![Figure][11]Mutations and deletions in the spike proteinCurrently, B.1.1.7, B.1.351, and P.1 are the major circulating variants of severe acute respiratory syndrome coronavirus 2 (SARS-CoV-2); others are emerging. The spike S1 subunit contains an amino (N)-terminal domain (NTD) and receptor-binding domain (RBD), which mediate host receptor recognition and contain epitopes for antibody binding. Deletions and substitutions in S1 can affect transmissibility (Tr), vaccine efficacy (Ef), and virulence (Vi). Additional mutations that define the variants can be tracked at ([ 8 ][7]). SP, signal peptide.GRAPHIC: N. CARY/ SCIENCE It is not yet known whether the complex mutational patterns observed in SARS-CoV-2 variants are linked on the same viral genome or represent mixtures of different variants within the same patient. Studies evaluating the linkage of these mutations in individual SARS-CoV-2 genomes using single-genome amplification and sequencing, as has been used to characterize genetic diversity of HIV-1 and other viruses, are needed to accurately assess the infectivity and phenotype of individual variants. A case report of intrahost SARS-CoV-2 evolution showed that SARS-CoV-2 can evolve multiple distinct lineages within the same individual ([ 6 ][6]).Several studies suggest that the major circulating variants have reduced neutralizing sensitivity to CP and plasma from recently vaccinated individuals. For example, CP from individuals who were infected with the B.1 lineage (D614G-containing SARS-CoV-2) had varying reductions in neutralizing activity against live virus isolates of the B.1.351 lineage. Additionally, vaccine-elicited antibodies have reduced neutralization of pseudovirus containing the triple mutation in S1 (K417N, E484K, and N501Y) of the P.1 and B.1.351 variants ([ 12 ][12]). Pseudovirus bearing the deletions and mutations found in the B.1.1.7 lineage also showed reduced neutralization sensitivity, but titers of antibody were sufficient for complete neutralization of B.1.1.7 in sera from 40 individuals vaccinated with BNT162b2 (Pfizer/BioNTech) ([ 13 ][13]). Continued phenotypic assessments of emerging, rapidly spreading variants, including those with nonsynonymous mutations in S1 (NTD and RBD) and S2, to neutralization by CP and postvaccination sera should be a high priority to monitor possible effects on vaccine efficacy.Phase 3 trials of SARS-CoV-2 vaccines derived from a single S sequence have shown them to be highly effective in preventing infection with the initial SARS-CoV-2 variants, including those with the D614G mutation ([ 14 ][14], [ 15 ][15]). More recent data suggest that certain vaccines are less protective against the B.1.351 variant, although additional studies are needed. Studies showing reduced antibody sensitivities against new variants do not inherently prove that a vaccine is less effective. In addition to effector B cells (which produce neutralizing antibodies), there are numerous additional vaccine-induced responses of the innate and adaptive immune systems that may protect against infection and further viral immune escape. Conversely, there are uncharacterized mutations outside of S that could facilitate SARS-CoV-2 immune evasion.The growing evidence for the emergence of immune escape mutations in protracted SARS-CoV-2 infection and for multiple, rapidly spreading variants should raise broad concern and action. Reducing the spread of SARS-CoV-2 is most likely to prevent further selection of immune escape variants. This will require a coordinated and comprehensive global vaccination and prevention strategy. Partial roll-out and incomplete immunization of individuals leading to suboptimal titers of neutralizing antibody could promote selection of escape variants that negatively affect vaccine efficacy. Increased genotypic and phenotypic testing capacities are essential worldwide to detect and characterize circulating SARS-CoV-2 variants that may emerge from selection by natural or vaccine-mediated immune responses. Infections that occur among vaccinated individuals should be aggressively evaluated for the mechanisms of breakthrough. The explosive, global spread of SARS-CoV-2 and the devastation it has wreaked is a stark warning of the potential for new variants to further complicate pandemic control. Vaccine manufacturers are now testing potential booster vaccines against circulating SARS-CoV-2 variants, and more broadly active monoclonal antibodies are in development for therapy. Such proactive approaches are likely to be needed to ensure pandemic control and elimination.1.  [↵][16]1.   B. Dearlove et al        ., Proc. Natl. Acad. Sci. U.S.A. 117, 23652 (2020).        [OpenUrl][17][Abstract/FREE Full Text][18]2.  [↵][19]1.   N. M. A. Okba et al        ., Emerg. Infect. Dis. 26, 1478 (2020).        [OpenUrl][20][PubMed][21]3.  [↵][22]1.   V. A. Avanzato et al        ., Cell 183, 1901 (2020).        [OpenUrl][23][PubMed][21]4.  1.   J. H. Baang et al        ., J. Infect. Dis. 223, 23 (2021).        [OpenUrl][24]5.  [↵][25]1.   B. Choi et al        ., N. Engl. J. Med. 383, 2291 (2020).        [OpenUrl][26][PubMed][21]6.  [↵][27]1.   M. K. Hensley et al        ., Clin. Infect. Dis. 10.1093/cid/ciab072 (2021).        7.  [↵][28]1.   S. A. Kemp et al        ., Nature (2021). 10.1038/s41586-021-03291-y        8.  [↵][29]1.   A. H. O'Toole et al        ., International Lineage Report. 2021, PANGO lineages,         9.  [↵][30]1.   T. N. Starr et al        ., Cell 182, 1295 (2020).        [OpenUrl][31]10. [↵][32]1.   H. Gu et al        ., Science 369, 1603 (2020).        [OpenUrl][33][Abstract/FREE Full Text][34]11. [↵][35]Public Health England, Investigation of novel SARS-CoV-2 variant: Variant of Concern 202012/01 Technical Briefing 5 (2021); .        12. [↵][36]1.   Z. Wang et al        ., Nature 10.1038/s41586-021-03324-6 (2021).        13. [↵][37]1.   A. Muik et al        ., Science 371, eabg6105 (2021).        [OpenUrl][38]14. [↵][39]1.   F. P. Polack et al        ., N. Engl. J. Med. 383, 2603 (2020).        [OpenUrl][40][CrossRef][41][PubMed][21]15. [↵][42]1.   L. R. Baden et al        ., N. Engl. J. Med. 384, 403 (2021).        [OpenUrl][43][CrossRef][44][PubMed][21]Acknowledgments: We thank L. Pollini for assistance. J.W.M. is a consultant to Gilead Sciences and holds shares or share options in Co-Crystal Pharma, Inc., Abound Bio, Inc., and Infectious Disease Connect. [1]: #ref-1 [2]: #ref-2 [3]: #ref-3 [4]: #ref-7 [5]: #ref-5 [6]: #ref-6 [7]: #ref-8 [8]: #ref-9 [9]: #ref-10 [10]: #ref-11 [11]: pending:yes [12]: #ref-12 [13]: #ref-13 [14]: #ref-14 [15]: #ref-15 [16]: #xref-ref-1-1 "View reference 1 in text" [17]: {openurl}?query=rft.jtitle%253DProc.%2BNatl.%2BAcad.%2BSci.%2BU.S.A.%26rft_id%253Dinfo%253Adoi%252F10.1073%252Fpnas.2008281117%26rft_id%253Dinfo%253Apmid%252F32868447%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx [18]: /lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6NDoicG5hcyI7czo1OiJyZXNpZCI7czoxMjoiMTE3LzM4LzIzNjUyIjtzOjQ6ImF0b20iO3M6MjM6Ii9zY2kvMzcxLzY1MzYvMTMwNi5hdG9tIjt9czo4OiJmcmFnbWVudCI7czowOiIiO30= [19]: #xref-ref-2-1 "View reference 2 in text" [20]: {openurl}?query=rft.jtitle%253DEmerg.%2BInfect.%2BDis.%26rft.volume%253D26%26rft.spage%253D1478%26rft_id%253Dinfo%253Apmid%252Fhttp%253A%252F%252Fwww.n%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx [21]: /lookup/external-ref?access_num=http://www.n&link_type=MED&atom=%2Fsci%2F371%2F6536%2F1306.atom [22]: #xref-ref-3-1 "View reference 3 in text" [23]: {openurl}?query=rft.jtitle%253DCell%26rft.volume%253D183%26rft.spage%253D1901%26rft_id%253Dinfo%253Apmid%252Fhttp%253A%252F%252Fwww.n%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx [24]: {openurl}?query=rft.jtitle%253DJ.%2BInfect.%2BDis.%26rft.volume%253D223%26rft.spage%253D23%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx [25]: #xref-ref-5-1 "View reference 5 in text" [26]: {openurl}?query=rft.jtitle%253DN.%2BEngl.%2BJ.%2BMed.%26rft.volume%253D383%26rft.spage%253D2291%26rft_id%253Dinfo%253Apmid%252Fhttp%253A%252F%252Fwww.n%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx [27]: #xref-ref-6-1 "View reference 6 in text" [28]: #xref-ref-7-1 "View reference 7 in text" [29]: #xref-ref-8-1 "View reference 8 in text" [30]: #xref-ref-9-1 "View reference 9 in text" [31]: {openurl}?query=rft.jtitle%253DCell%26rft.volume%253D182%26rft.spage%253D1295%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx [32]: #xref-ref-10-1 "View reference 10 in text" [33]: {openurl}?query=rft.jtitle%253DScience%26rft_id%253Dinfo%253Adoi%252F10.1126%252Fscience.abc4730%26rft_id%253Dinfo%253Apmid%252F32732280%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx [34]: /lookup/ijlink/YTozOntzOjQ6InBhdGgiO3M6MTQ6Ii9sb29rdXAvaWpsaW5rIjtzOjU6InF1ZXJ5IjthOjQ6e3M6ODoibGlua1R5cGUiO3M6NDoiQUJTVCI7czoxMToiam91cm5hbENvZGUiO3M6Mzoic2NpIjtzOjU6InJlc2lkIjtzOjEzOiIzNjkvNjUxMS8xNjAzIjtzOjQ6ImF0b20iO3M6MjM6Ii9zY2kvMzcxLzY1MzYvMTMwNi5hdG9tIjt9czo4OiJmcmFnbWVudCI7czowOiIiO30= [35]: #xref-ref-11-1 "View reference 11 in text" [36]: #xref-ref-12-1 "View reference 12 in text" [37]: #xref-ref-13-1 "View reference 13 in text" [38]: {openurl}?query=rft.jtitle%253DScience%26rft.volume%253D371%26rft.spage%253Deabg6105%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx [39]: #xref-ref-14-1 "View reference 14 in text" [40]: {openurl}?query=rft.jtitle%253DN.%2BEngl.%2BJ.%2BMed.%26rft.volume%253D383%26rft.spage%253D2603%26rft_id%253Dinfo%253Adoi%252F10.1056%252FNEJMoa2034577%26rft_id%253Dinfo%253Apmid%252Fhttp%253A%252F%252Fwww.n%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx [41]: /lookup/external-ref?access_num=10.1056/NEJMoa2034577&link_type=DOI [42]: #xref-ref-15-1 "View reference 15 in text" [43]: {openurl}?query=rft.jtitle%253DN.%2BEngl.%2BJ.%2BMed.%26rft.volume%253D384%26rft.spage%253D403%26rft_id%253Dinfo%253Adoi%252F10.1056%252FNEJMoa2035389%26rft_id%253Dinfo%253Apmid%252Fhttp%253A%252F%252Fwww.n%26rft.genre%253Darticle%26rft_val_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Ajournal%26ctx_ver%253DZ39.88-2004%26url_ver%253DZ39.88-2004%26url_ctx_fmt%253Dinfo%253Aofi%252Ffmt%253Akev%253Amtx%253Actx [44]: /lookup/external-ref?access_num=10.1056/NEJMoa2035389&link_type=DOI</p><p>Warning: Can only detect less than 5000 characters</p></div><div id="story_share_this"><div class="sharethis-inline-share-buttons"></div></div><div class="text-break sotry_link page_narrow"><a target="_blank" href="https://science.sciencemag.org/content/371/6536/1306">https://science.sciencemag.org/content/371/6536/1306</a></div><div class="story_tags page_narrow"><button type="button" class="btn btn-light my_tag"><a href="/tag/出现/">#出现</a></button><button type="button" class="btn btn-light my_tag"><a href="/tag/warning/">#warning</a></button></div></div><div class="my_movie_list_item shadow p-3 mb-5 bg-white rounded"><button type="button" class="btn btn-link my_tag"><a href="/tag/web2.0/">#web2.0</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/google/">#google</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/设计/">#设计</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/创意/">#创意</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/摄影/">#摄影</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/图片/">#图片</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/游戏/">#游戏</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/软件/">#软件</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/视频/">#视频</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/手机/">#手机</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/广告/">#广告</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/iphone/">#iphone</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/网站/">#网站</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/免费/">#免费</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/下载/">#下载</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/windows/">#windows</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/微软/">#微软</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/firefox/">#firefox</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/blog/">#blog</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/苹果/">#苹果</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/apple/">#apple</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/音乐/">#音乐</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/博客/">#博客</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/wordpress/">#wordpress</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/恶搞/">#恶搞</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/qq/">#qq</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/艺术/">#艺术</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/web/">#web</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/工具/">#工具</a></button><button type="button" class="btn btn-link my_tag"><a href="/tag/分享/">#分享</a></button></div></div></div><div id="my_footer"><div class=""><a href="/tags/">tags</a> <a href="/users/">users</a></div>&copy;2012-2021 diglog.com </div></div></body></html>